Documents/Testing Policy/TestPolicy.tex

```diff
@@ -4,128 +4,308 @@
+\indent Tests for these functions can be found \href{https://github.com/BinaryNinjaz/COS301-Capstone/blob/master/Source/Web/functions/test.js}{here}.\\
+\indent These functions are tested simply by giving them dummy data and confirming that the correct results are returned.
+\subsection{Mobile Application and Website}
+These are umbrella terms that each encompass a variety of smaller subsystems.
+\subsection{Collection Tracker}
+The collection tracker (also termed as yield tracker) is the primary mobile activity performed by foremen. It is how they measure the collections of the workers.\\
+\indent On iOS this is tested by unit tests that create fake information and then feeds it into the tracker to confirm that arbitary information can be processed. Simultaneoulsy, it is confirmed that the location and time measured by the tracker is handled correctly. These tests can be found \href{https://github.com/BinaryNinjaz/COS301-Capstone/blob/iOS-/Source/master/Harvest/HarvestTests/HarvestTests.swift}{here}.\\
+\indent On Android, testing is carried out by means of UI tests which confirm that arrival to the tracker and its operations is carried out correctly as a whole. The tests can be found \href{https://github.com/BinaryNinjaz/COS301-Capstone/blob/master/Source/Android/Harvest/app/src/androidTest/java/za/org/samac/harvest/YieldTrackerTest.java}{here}.
+\subsection{Foreman Location Viewer}
+The foreman location viewer is used to display the live locations of requested foremen, it relies heavily on accurate location reporting from the foreman tracker. Once the information is received, it is simply displayed.
+\subsection{Foreman Tracker}
+The foreman tracker is used to report the location of the foreman (the mobile application) to the foreman location viewer. It relies almost exclusively on accurate location reporting, which is received from hardware and written to the database.
+\subsection{Collection Viewer}
+The collection viewer shows summaries of previously recorded sessions. It is used solely to report simple previously recorded information to the user. This information is read from the database, tests for which have already been discussed.
+\subsection{Statistics Viewer}
+The statistics viewer is used to report information to the user in the form of line graphs. Given the large amounts of data processed, this is handled by cloud functions, testing for which has already been discussed. The complication being the correct reporting of the information received.\\
+\indent Android unit tests, which confirm the correct handling of dates can be found \href{https://github.com/BinaryNinjaz/COS301-Capstone/blob/Android/Source/master/Harvest/app/src/test/java/za/org/samac/harvest/StatsTests.java}{here}.
+\subsection{Information Manager}
```

Comment:
Tests [here](https://github.com/BinaryNinjaz/COS301-Capstone/blob/525c530629e004b2ad3c382646f4d7f100349adc/Source/iOS/Harvest/HarvestTests/HarvestTests.swift#L205) and  downwards